File: workspace/Notification/src/main/java/br/com/cers/notification/service/NotificationService.java
package br.com.cers.notification.service;

import org.springframework.data.domain.Page;

import br.com.cers.notification.entity.Notification;

public interface NotificationService {

	Notification findByTitle(String title);

	Notification createOrUpdate(Notification notification);

	Notification findByIdNotification(Integer idNotification);

	void delete(Integer idNotification);

	Page<Notification> findAll(int page, int count);

}

File: cersNotification/src/app/services/notification.service.ts
import { Notificationn } from './../model/notificationn.model';
import { Injectable } from '@angular/core';
import { NOTIFICATION_API } from './notification.api';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class NotificationService {

  constructor(private http: HttpClient) { }

  createOrUpdate(notificationn: Notificationn) {
    if(notificationn.idnotification != null && notificationn.idnotification != '') {
      return this.http.put(`${NOTIFICATION_API}/api/notification`, notificationn);
    }
    return this.http.post(`${NOTIFICATION_API}/api/notification`, notificationn);
  }

  findAll(page:number, count: number) {
    return this.http.get(`${NOTIFICATION_API}/api/notification/${page}/${count}`);
  }

  findById(id:string) {
    return this.http.get(`${NOTIFICATION_API}/api/notification/${id}`);
  }
  
  delete(id:string) {
    return this.http.delete(`${NOTIFICATION_API}/api/notification/${id}`);
  }

}
